typespec-tests/src/main/java/com/cadl/patch/models/Resource.java

```diff
@@ -0,0 +1,355 @@
+     */
+    @Generated
+    public Integer getIntValue() {
+        return this.intValue;
+    }
+
+    /**
+     * Set the intValue property: The intValue property.
+     * 
+     * @param intValue the intValue value to set.
+     * @return the Resource object itself.
+     */
+    @Generated
+    public Resource setIntValue(Integer intValue) {
+        this.intValue = intValue;
+        this.updatedProperties.add("intValue");
+        return this;
+    }
+
+    /**
+     * Get the enumValue property: The enumValue property.
+     * 
+     * @return the enumValue value.
+     */
+    @Generated
+    public ResourceEnumValue getEnumValue() {
+        return this.enumValue;
+    }
+
+    /**
+     * Set the enumValue property: The enumValue property.
+     * 
+     * @param enumValue the enumValue value to set.
+     * @return the Resource object itself.
+     */
+    @Generated
+    public Resource setEnumValue(ResourceEnumValue enumValue) {
+        this.enumValue = enumValue;
+        this.updatedProperties.add("enumValue");
+        return this;
+    }
+
+    /**
+     * Get the innerModelProperty property: The wireNameForInnerModelProperty property.
+     * 
+     * @return the innerModelProperty value.
+     */
+    @Generated
+    public InnerModel getInnerModelProperty() {
+        return this.innerModelProperty;
+    }
+
+    /**
+     * Set the innerModelProperty property: The wireNameForInnerModelProperty property.
+     * 
+     * @param innerModelProperty the innerModelProperty value to set.
+     * @return the Resource object itself.
+     */
+    @Generated
+    public Resource setInnerModelProperty(InnerModel innerModelProperty) {
+        this.innerModelProperty = innerModelProperty;
+        this.updatedProperties.add("innerModelProperty");
+        return this;
+    }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        if (jsonMergePatch) {
+            return toJsonMergePatch(jsonWriter);
+        } else {
+            jsonWriter.writeStartObject();
+            jsonWriter.writeMapField("map", this.map, (writer, element) -> writer.writeJson(element));
+            jsonWriter.writeStringField("description", this.description);
+            jsonWriter.writeNumberField("longValue", this.longValue);
+            jsonWriter.writeNumberField("intValue", this.intValue);
+            jsonWriter.writeStringField("enumValue", this.enumValue == null ? null : this.enumValue.toString());
+            jsonWriter.writeJsonField("wireNameForInnerModelProperty", this.innerModelProperty);
+            return jsonWriter.writeEndObject();
+        }
+    }
+
+    public JsonWriter toJsonMergePatch(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        if (this.map != null) {
+            jsonWriter.writeMapField("map", this.map, (writer, element) -> {
+                element.serializeAsJsonMergePatch(true);
+                writer.writeJson(element);
```

Comment:
Set serializeAsJsonMergePatch back to `false` after write to json.

Reply:
Here you need null check and write null value. You can do it in follow-up PR.

Reply:
`writer.writeJson(element);` will call `element.toJson()`, and in `toJson()` will call `toJsonMergePatch()` which will handle the null values.
Test case: `testSerializationForNestedModelProperty`

Reply:
I mean `element` be null.

Reply:
oh, I see. This is already fixed in this pr. output code: https://github.com/Azure/autorest.java/pull/2424/files#diff-0318904499f974c90db93a7e3663c59db25ffcb359ea2f7081a2d1b5e3b748d7